api/src/main/java/module-info.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2022 International Business Machines Corp. and others
+ * 
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership. Licensed under the Apache License, 
+ * Version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+module jakarta.batch {
+
+    requires jakarta.inject;
+    requires java.logging;
+    requires jakarta.cdi;
```

Comment:
shouldn't it be `requires static` instead of `requires` for cdi and logging?

Reply:
* logging - Not sure how critical this is.  In a sense it really should be a runtime dependency since [BatchRuntime](https://github.com/eclipse-ee4j/batch-api/blob/d49675492b5f1f77f2d53236e87d6f7ee1bd8f78/api/src/main/java/jakarta/batch/runtime/BatchRuntime.java#L49-L51) uses java.util.logging.   (This isn't part of the spec, but of the API jar mgd by the spec project.)   But I think using an optional (requires static) dependency might actually also work, since I'm not sure how to create a runtime where java.logging is not available.
* cdi - It sounds right to me to make this optional.  Even though implementers would have to add it....there's enough usefulness without CDI that I suppose that's a good tradeoff, since it brings in a lot of extra deps.
* inject - In a sense this COULD be left as optional like 'cdi'.  If you had just a "client" running jobs and checking status you wouldn't need inject necessarily.   Going to try to spend a bit more time investigating this.

Also I need to add:
> uses jakarta.batch.operations.JobOperator;

Reply:
I pushed new commits with **jakarta.cdi** optional and **jakarta.logging** required.

That's all that's needed to add JPMS to the status quo.    

Both of the points of: 
1. removing logging from BatchRuntime
2. failing if a second JobOperator is detected via ServiceLoader
are separate issues. 

IMHO, they don't need to be resolved in this release but feel free to open them if you want.

Note on 2., that [CDI](https://github.com/eclipse-ee4j/cdi/blob/4507fa89b170044a35b31471176ea81bcb2599fa/api/src/main/java/jakarta/enterprise/inject/spi/CDI.java#L96-L100) seems to have an example where they simply use the first service impl found.

Reply:
@tjwatson I can't mark you as reviewer but if you don't mind taking a quick look as a JPMS expert . .. thank you :)

Reply:
@scottkurz hehe, you are naughty with this reference but nah :p. Joke apart `checkProvider` does the drop of the "not current" CDI instances so it is not random/the first one, it is the "first one active" which also means "the only one".

Reply:
Glad I made you laugh there ;) Well, still, we've had it this way for nine years and no one complained.   Maybe because you're going to collide on the CDI producers before you can collide with JobOperator.

Reply:
Well, I can admit I'm maybe not the best representative user but I hit this issue quite a few - always worked around with a custom classloader faking a single SPI registration but it is harder now with JPMS and it requires some deeper bytecode hack/injection :(. It happens easily if you run in a container with one impl - let say jberet - and want to force another one (batchee in my case).

But agree it does not urge for this release but we should ensure it is not random at least.

Reply:
>  worked around with a custom classloader faking a single SPI registration... if you run in a container with one impl - let say jberet - and want to force another one 

If we're just going to fail fast when we detect >1 JobOperator service impl then that's not going to make this use case any easier obviously.

I do think we're not making this problem any worse, so I opened https://github.com/eclipse-ee4j/batch-api/issues/193 to follow-up on this issue... probably for a future release.